packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -376,34 +434,39 @@ class FlutterErrorBuilder {
       level: level,
     ));
   }
-
+  
+  // TODO(inmatrix): ask what this method is for.
   void addBlock(String name, FlutterErrorBuilder blockContents) {
     throw 'XXX implement';
   }
 
-  /// Use this method to describe a [StackTrace] relevant to the error.
-  ///
-  /// There won't be
+  /// Adds a [StackTrace] relevant to the error.
   void addStackTrace(String name, StackTrace stackTrace, {IterableFilter<String> stackFilter}) {
     _parts.add(DiagnosticsStackTrace(name, stackTrace, stackFilter: stackFilter));
   }
 
+  /// Adds an [IntProperty]
   void addIntProperty(String name, int value) {
     _parts.add(IntProperty(name, value));
   }
 
+  /// Adds an instance of [DiagnosticNode] to the error report.
```

Comment:
we should probably indicate that generally the other methods should be used instead of this escape hatch as the other methods will by default use styles consistent with other error displays. For example, indenting values to show on the next line, etc.

Reply:
What do you mean by "other error displays"?

Reply:
The idea I'm trying to get at is that if you use this builder class your error will tend to be styled like other flutter errors. If you create a bunch of custom diagnostics you could end up with an error with inconsistent style due to arbitrary style choices such as whether property names and values should be on the same line and how much context you include when displaying a render object (e.g. showing the immediate properties, a truncated tree, or the full tree).